src/mlpack/core/dists/gaussian_distribution.hpp

```diff
@@ -103,7 +103,24 @@ class GaussianDistribution
    * @param logProbabilities Output log probabilities for each input
    *     observation.
    */
-  void LogProbability(const arma::mat& x, arma::vec& logProbabilities) const;
+  void LogProbability(const arma::mat& x, arma::vec& logProbabilities) const
+  {
+    // Column i of 'diffs' is the difference between x.col(i) and the mean.
+    arma::mat diffs = x - (mean * arma::ones<arma::rowvec>(x.n_cols));
```

Comment:
I wonder if ``arma::mat diffs = x.each_col() - mean`` is something that would improve the readability and or the speed.

Reply:
@zoq I think that would improve readability. However, I am not quite sure about speed. Do you wish to know the results? I can try to run tests in my local server and check it out.

Reply:
Should we leave the implementation as it is?

Reply:
I would give ``each_col()`` a try.